bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java

```diff
@@ -328,6 +334,9 @@ public void readComplete(int rc, LedgerHandle lh,
                 if (rc != BKException.Code.OK) {
                     LOG.error("BK error reading ledger entry: " + entryId,
                             BKException.create(rc));
+                    if (entriesUnableToReadForThisLedger != null) {
```

Comment:
I would require that this variable is already non null, in order to not miss any possible code path that reaches this point